src/services/LiqualityService.ts

```diff
@@ -13,34 +13,23 @@ import { EnvironmentAccessorService } from '@/services/enviroment-accessor.servi
 export default class LiqualityService extends WalletService {
   private bitcoinProvider!: WindowBitcoinProvider;
 
-  private enabled = false;
-
   constructor(testBitcoinProvider?: WindowBitcoinProvider) {
     super();
     if (testBitcoinProvider) {
       this.bitcoinProvider = testBitcoinProvider;
-      this.enabled = true;
     }
   }
 
   private enable(): Promise<void> {
     return new Promise<void>((resolve, reject) => {
-      const w = window as Window;
-      if (!this.enabled) {
-        try {
-          if (w.bitcoin) {
-            this.bitcoinProvider = window.bitcoin;
-            this.bitcoinProvider.enable()
-              .then(([address]: LiqualityAddress[]) => {
-                this.enabled = !!address;
-                resolve();
-              });
-          }
-        } catch (e) {
-          reject(e);
-        }
-      } else {
-        resolve();
+      try {
+        this.bitcoinProvider = window.bitcoin;
+        window.bitcoin.enable()
```

Comment:
New concern here. The way it is done now it seems like calling `enable()` multiple times will initiate a new call to `liquality.enable()`. According to their documentation https://docs.liquality.io/wallet/bitcoin-api/getting-started#enable this prompts the user to accept the application, what happens if we call this method multiple times? is it going to prompt the user each time?
I think the pre existing `enabled` field made sense, at the same time I can see that if the `enable` method does not prompt the user, it should not harm.
Could you please clarify this doubt?

Reply:
If the user has already accepted the application, the next call to liquality.enable() don't show liqualitiy's popup again. I tried many situations and I didn't find any problem related to this question.